### DISCUSSION TITLE
Adding custom classes

### DISCUSSION THREAD

**Discussion by artello73** [ssssota/svelte-exmarkdown#214] [General]
Hello,

Facing a need to classes to HTML tags 
in general 

```
# Hello, world
```
needs to be transformed to 

```html
<h1 class="h1"> Hello, world</h1>
```

There is a rehype plugin available, which seems do the trick https://github.com/riderjensen/rehype-class-names
but, I have a difficult time to understand how to connect it. 
I tried following: 
```typescript
import type { Plugin } from 'svelte-exmarkdown';
import rehypeClassNames from 'rehype-class-names';
const addClass: Plugin = {
	rehypePlugin: [
		rehypeClassNames,
		{
			h1: 'h1',
			h2: 'h2'
		}
	]
};
```
but it causing bunch of TypeScript errors .... Any ideas?

**Comment by ssssota** [comment]
Because [rehype-class-names](https://github.com/riderjensen/rehype-class-names) has an old `unified` package (v10).
svelte-exmarkdown uses `unified` v11.

You can ignore the type error.
If you feel bad, you can post an issue or PR to rehype-class-names to update unified.

**Reply by artello73** [reply]
Yes, you are right! Added the plugin locally with updated v11 unified, and got working with very minor changes!
Thanks a lot!

**Comment by HuakunShen** [comment]
Thanks I got the custom class working with this rehype plugin.
Another question, how to add a copy button to the code blocks.

**Reply by HuakunShen** [reply]
NVM I found this issue https://github.com/ssssota/svelte-exmarkdown/issues/258
